doc/ni_by_example_4.md

```diff
@@ -0,0 +1,169 @@
+* `tpe`: time parts to epoch
+  * `tpe(@time_pieces)`: Returns the epoch time and assumes that the pieces are year, month, day, hour, minute, and second, in that order.
+  * `tpe($time_format, @time_pieces)`: Returns the epoch time, using `$time_format` to determine what the ordered `@time_pieces` are.
+* `tep`: time epoch to parts
+  * `tep($epoch_time)`: returns the year, month, day, hour, minute, and second in human-readable formatfrom the epoch time.
+  * `tep($time_format, $epoch_time)`: returns the specified parts of the date using following `$time_format`.
+* `timezone_seconds`
+  * `tep($raw_timestamp + $timezone_seconds($lat, $lng))` returns the approximate date and time at the location `$lat, $lng` at a Unix timestamp of `$raw_timestamp`.
+  
+  
+##Plotting with `ni --js`
+Check out the [tutorial](tutorial.md) for some examples of cool, interactive `ni` plotting.
+
+**TODO**: Say something useful.
+
+
+##Custom Compound Reduce
+#### `rfn`: Custom compound reduce
+
+**TODO: Understand this**
+
+##Matrix Operations
+
+Operations on huge matrices are not entirely `ni`ic, since they may require space greater than memory, whichwill make them slow. However, operators are provided to improve These operations are suited best to 
+
+
+* `X<col>`, `Y<col>`, `N<col>`: Matrix Partitioning
+  * **TODO**: understand how these actually work.
+* `X`: sparse-to-dense transformation
+  * In the case that there are collisions for locations `X`, `X` will sum the values
+  * For example: `ni n010p'r 0, a%3, 1' X`
+
+##Disk-Backed Data Closures
+
+* `@:[disk_backed_data_closure]`
+
+##Annoyingly Advanced Perl
+* `use strict` and the `::` prefix in a Perl Environment
+  * When `use strict` is enabled, Perl will complain when you try to create a variable in a Perl snippet that does not start with `::`.
+  * The reasons for this are very specific to Perl; if you are a true Perl nerd, you can look them up, but you do not need to know them if you just accept that variables need to start with `::` when you apply `use strict`.
+  * It is probably a good idea to `use strict` when the variables you define are sufficiently complex; otherwise you're probably okay not using it.
+  
+##Binary Operations
+In theory, this can save you a lot of space. But I haven't used this in practice.
+  
+
+##Less Useful `ni`-specific Perl Extensions
+
+
+###Array Functions
+  * `clip`
+  * `within`
+  
+##Stream Splitting/Joining/Duplication
+I don't find these operators particularly useful in practice (with the exception of `=\>` for writing a file in the middle of a long processing pipeline), but it's possible that you will! Then come edit these docs and explain why.
+
+* `+`: append a stream to this one
+* `^`: prepend a stream to this one
+* `=`: duplicate this stream through a process, discarding its output
+   
+##Writing Your Own `ni` Extensions
+**TODO** Understand how this works
```

Comment:
You should probably hold off on documenting this because I'm about to rewrite all of the internals unrecognizably.